File: 15 labb/lab15.py
import math

def f(x):
    return (5 * x ** 2 - 8) / (x ** 3 + 1)

def g(x):
    return math.sqrt(x + 1) - math.sqrt(x) - (1 / 2)

m = 10000
n = 1000

A = [[f(i) + g(j) for j in range(1, m + 1)] for i in range(1, n + 1)]

N_A = math.sqrt(n * m) * max(abs(A[i][j]) for i in range(n) for j in range(m))

print(f'{N_A:.6f}')

File: 15 labb/lab15.c
#include <stdio.h>
#include <math.h>
#include <time.h>

double f (double x)
{
    return (5 * pow(x, 2) - 8) / (pow(x, 3) + 1);
}

double g (double x)
{
    return sqrt(x + 1) - sqrt(x) - 0.5;
}

int main()
{
    int n, m, i, j;

    n = 10000;
    m = 1000;

    double max_element = 0;
    double N_A;
    double A[n][m];
    
    clock_t begin = clock();

    for (i = 0; i < n; i++)
    {
        for (j = 0; j < m; j++)
        {
            A[i][j] = (f(i + 1) + g(j + 1));
            if (fabs(A[i][j]) > max_element)
            {
                max_element = fabs(A[i][j]);
            }
        }
    }

    N_A = sqrt(n * m) * max_element;

    clock_t end = clock();

    double time_spent = (double)(end - begin) / CLOCKS_PER_SEC;

    printf("результат: %.6f\n", N_A);
    printf("время: %.6f\n", time_spent);
    return 0;
}